crates/egui_demo_lib/src/demo/multi_touch.rs

```diff
@@ -83,21 +101,39 @@ impl crate::View for MultiTouch {
             // check for touch input (or the lack thereof) and update zoom and scale factors, plus
             // color and width:
             let mut stroke_width = 1.;
-            if let Some(multi_touch) = ui.ctx().multi_touch() {
-                // This adjusts the current zoom factor and rotation angle according to the dynamic
-                // change (for the current frame) of the touch gesture:
-                self.zoom *= multi_touch.zoom_delta;
-                self.rotation += multi_touch.rotation_delta;
-                // the translation we get from `multi_touch` needs to be scaled down to the
-                // normalized coordinates we use as the basis for painting:
-                self.translation += to_screen.inverse().scale() * multi_touch.translation_delta;
-                // touch pressure will make the arrow thicker (not all touch devices support this):
-                stroke_width += 10. * multi_touch.force;
-
-                self.last_touch_time = ui.input(|i| i.time);
+            if ui.input(|i| i.multi_touch().is_some()) || relative_pointer_gesture {
+                ui.input(|input| {
+                    // This adjusts the current zoom factor, rotation angle, and translation according
+                    // to the dynamic change (for the current frame) of the touch gesture:
+                    self.zoom *= input.zoom_delta();
+                    self.rotation += input.rotation_delta();
+                    self.translation += to_screen.inverse().scale()
+                        * input
+                            .multi_touch()
+                            .map_or(input.smooth_scroll_delta, |mt| mt.translation_delta);
```

Comment:
I think it would make sense to have a common interface for this, like we do for `.zoom_delta()` and `.rotation_delta()`:


```suggestion
                    self.translation += to_screen.inverse().scale() * input.pan_delta();
```

Reply:
Would that only have a version using `smooth_scroll_delta`, or should it have a `raw_scroll_delta` variant as well?

Reply:
Let's just make one that uses `smooth_scroll_delta` - it's what most people should be using